[IMP] marketing automation: audience targeting

samueljlieber opened the pull request:
Task: #3774153

This PR improves the `marketing_automation/target_audience.rst` documentation by documenting the steps to target an audience for a marketing automation campaign, specifically detailing how a domain is defined with an example.

This PR can be FWP up to `master`.

robodoo commented:
[Pull request status dashboard](https://mergebot.odoo.com/odoo/documentation/pull/8018).

samueljlieber commented:
Hi @ksc-odoo this PR is ready for a peer review when you get the chance, thank you in advance!

ksc-odoo commented on content/applications/marketing/marketing_automation/target_audience.rst:
```diff
@@ -1,123 +1,123 @@
 ==================
-Target an audience
+Audience targeting
```
I could be mistaken, but I believe we're trying to avoid any "ing" phrasing in the main doc h1 -- so, perhaps the original h1 would work best? Again, things may have changed since that information was shared with me. So, this title may work. worth checking into, either way. figured it was worth leaving a comment about. 👍

samueljlieber replied:
This was [a suggestion](https://github.com/odoo/documentation/pull/7711#discussion_r1501237914) from ZST in my previous PR, I'll discuss and update as needed :)

samueljlieber replied:
@ksc-odoo I spoke with ZST and you are correct that we're trying to avoid any "ing" phrasing in the h1, however, in this case "Audience targeting" is an industry term and is more recognizable than "Target an audience". 

So for the benefit of the _target audience_ 🤓 of this doc, I am going to leave the H1 as "Audience targeting" :)

ksc-odoo commented on content/applications/marketing/marketing_automation/target_audience.rst:
```diff
@@ -1,123 +1,123 @@
-When configuring the target filters on a marketing campaign, there are some options that have an
-:guilabel:`> (arrow)` icon beside them. The :guilabel:`> (arrow)` icon signifies that the particular
-filter has more refined parameters within it that can be customized.
+This section can be found on a campaign form by selecting an existing campaign or :ref:`creating a
```
```suggestion
This section can be found on a campaign form by selecting an existing campaign, or :ref:`creating a
```

ksc-odoo commented on content/applications/marketing/marketing_automation/target_audience.rst:
```diff
@@ -1,123 +1,123 @@
-:guilabel:`> (arrow)` icon beside them. The :guilabel:`> (arrow)` icon signifies that the particular
-filter has more refined parameters within it that can be customized.
+This section can be found on a campaign form by selecting an existing campaign or :ref:`creating a
+new campaign <marketing_automation/campaigns>` from the :menuselection:`Marketing Automation -->
```
```suggestion
new campaign <marketing_automation/campaigns>` from the :menuselection:`Marketing Automation app -->
```

ksc-odoo commented on content/applications/marketing/marketing_automation/target_audience.rst:
```diff
@@ -1,123 +1,123 @@
+  determines the fields that are available throughout the campaign, including the fields available
+  in the :guilabel:`Filter` section, and in dynamic placeholders.
+- :guilabel:`Save as Favorite Filter`: saves the current :guilabel:`Filter` for future use with the
+  current :guilabel:`Target` model, and can be managed from the :menuselection:`Marketing Automation
```
```suggestion
  current :guilabel:`Target` model, and can be managed from the :menuselection:`Marketing Automation app
```

ksc-odoo commented on content/applications/marketing/marketing_automation/target_audience.rst:
```diff
@@ -1,123 +1,123 @@
+- :guilabel:`Save as Favorite Filter`: saves the current :guilabel:`Filter` for future use with the
+  current :guilabel:`Target` model, and can be managed from the :menuselection:`Marketing Automation
+  --> Configuration --> Favorite Filters` menu.
+- :guilabel:`Unicity based on`: specifies the :guilabel:`Target` model field at which to avoid
+  duplicates by. Traditionally, the :guilabel:`Email` field is used, but any available field can be
+  used.
```
```suggestion
- :guilabel:`Unicity based on`: specifies the :guilabel:`Target` model field at which duplicates
  should be avoided. Traditionally, the :guilabel:`Email` field is used, but any available field can
  be used.
```

ksc-odoo commented on content/applications/marketing/marketing_automation/target_audience.rst:
```diff
@@ -1,123 +1,123 @@
-Then, add another node to this filter by clicking the :guilabel:`➕ (plus sign)` icon to the right of
-the equation.
+.. note::
+   Each activity in a campaigns workflow can target a subset of the target audience, see the
```
```suggestion
   Each activity in a campaign's workflow can target a subset of the target audience, see the
```

ksc-odoo commented on content/applications/marketing/marketing_automation/target_audience.rst:
```diff
@@ -1,123 +1,123 @@
 
-Once again, click the default :guilabel:`🟰 (equal sign)` icon in the second portion of the filter
-equation, and select :guilabel:`contains`.
+The default campaign :guilabel:`Filter` configuration is to :guilabel:`Match all records`,
```
```suggestion
The default campaign :guilabel:`Filter` configuration is set to :guilabel:`Match all records`,
```

ksc-odoo commented on content/applications/marketing/marketing_automation/target_audience.rst:
```diff
@@ -1,123 +1,123 @@
+At the top-left of the filter rules is a drop-down menu with two options. The selected option
+determines the criteria of how all of the rules are combined to filter the target audience:
+
+- :guilabel:`Match all 🔽 of the following rules`: all of the filter rules must be met for a record
```
```suggestion
- :guilabel:`Match all 🔽 of the following rules`: **all** of the filter rules must be met for a record
```

ksc-odoo commented on content/applications/marketing/marketing_automation/target_audience.rst:
```diff
@@ -1,123 +1,123 @@
+
+- :guilabel:`Match all 🔽 of the following rules`: all of the filter rules must be met for a record
+  to be targeted by the campaign. Logically, this is an *AND* (`&`) operation.
+- :guilabel:`Match any 🔽 of the following rules`: any of the filter rules can be met for a record
```
```suggestion
- :guilabel:`Match any 🔽 of the following rules`: **any** of the filter rules can be met for a record
```

ksc-odoo commented on content/applications/marketing/marketing_automation/target_audience.rst:
```diff
@@ -1,123 +1,123 @@
+   value. The :ref:`available conditional operators <reference/orm/domains>` are specific to the
+   field's data type.
+#. The third inline input is the variable *value* of the field name. The value input may appear as a
+   drop-down menu, a text input, a number input, a date/time input, a boolean selector, or blank
```
```suggestion
   drop-down menu, a text input, a number input, a date/time input, a boolean selector, or it may be blank,
```

ksc-odoo commented on content/applications/marketing/marketing_automation/target_audience.rst:
```diff
@@ -1,123 +1,123 @@
+#. :guilabel:`➕ (plus sign)`: adds a new rule below the existing rule.
+#. :guilabel:`(Add branch)`: adds a new group of rules below the existing rule, with the
+   :guilabel:`any` and :guilabel:`all` matching options available to define how each rule within
+   this branch is applied to filter. If the matching option is set to the same as the parent group,
```
```suggestion
   this branch is applied to the filter. If the matching option is set to the same as the parent group,
```

ksc-odoo commented on content/applications/marketing/marketing_automation/target_audience.rst:
```diff
@@ -1,123 +1,123 @@
+   this branch is applied to filter. If the matching option is set to the same as the parent group,
+   the fields are moved to join the parent group.
+#. :guilabel:`🗑️ (garbage can)`: deletes the node. If a branch node is deleted, all children of
+   that node are deleted as well.
```
```suggestion
   that node are deleted, as well.
```

ksc-odoo commented on content/applications/marketing/marketing_automation/target_audience.rst:
```diff
@@ -1,123 +1,123 @@
+
+At the bottom of the filter rules is a :guilabel:`# record(s)` button, which indicates the total
+number of records targeted by this domain. Select the :guilabel:`# record(s)` button to open a
+:guilabel:`Selected records` pop-up and view the targeted records.
```
```suggestion
:guilabel:`Selected records` pop-up window. in which the targeted records can be viewed.
```

ksc-odoo commented on content/applications/marketing/marketing_automation/target_audience.rst:
```diff
@@ -1,123 +1,123 @@
+
+.. example::
+   To target all leads and opportunities from the *CRM* app that are in the *New* stage, and have an
+   expected revenue greater than $1,000:
```
```suggestion
   expected revenue greater than $1,000, the following should be entered:
```

ksc-odoo reviewed (approved):
Alright @samueljlieber -- just finished my peer review on this. awesome job! nicely done! only a handful of super minor things that require your attention. Once you make the necessary adjustments, I think you can feel free to move this to the next stage. Way to go! 👍

StraubCreative commented on content/applications/marketing/marketing_automation/target_audience.rst:
```diff
@@ -1,123 +1,123 @@
-The Odoo *Marketing Automation* application helps marketers to do just that by providing detailed
-filtering tools, which can be as simple (or as complex) as necessary, to reach the right customers
-at the right time.
+The *Target* and *Filter* section of the campaign form, also referred to as the domain, contains the
+fields used to define the target audience for the campaign's reach (i.e., the unique contact records
+in the database).
```
- would use guis here since we're talking about fields on the UI, otherwise uncapitalized italics since they're not proper nouns
- wording: these are fields on a form, not sections as would be indicated by headings
- it's also know clear what exactly the *domain* is referring to: one field in particular or both of them together?
```suggestion
The :guilabel:`Target` and :guilabel:`Filter` fields on the campaign form, also referred to as the *domain*, contain the
parameters used to define the target audience for the campaign's reach (i.e., the unique contact records
in the database, and imported list, etc.).
```

StraubCreative commented on content/applications/marketing/marketing_automation/target_audience.rst:
```diff
@@ -1,123 +1,123 @@
+The target audience specifies the type of records available for use in the campaign, such as
+*Lead/Opportunity*, *Event Registration*, *Contact*, and more.
 
-When configuring the target filters on a marketing campaign, there are some options that have an
-:guilabel:`> (arrow)` icon beside them. The :guilabel:`> (arrow)` icon signifies that the particular
-filter has more refined parameters within it that can be customized.
+This section can be found on a campaign form by selecting an existing campaign, or :ref:`creating a
+new campaign <marketing_automation/campaigns>` from the :menuselection:`Marketing Automation app -->
+Campaigns` dashboard.
```
Don't need lines 9-14; they almost distract away from the small form coverage you're doing in the intro and in the list below.
```suggestion
```

StraubCreative commented on content/applications/marketing/marketing_automation/target_audience.rst:
```diff
@@ -1,123 +1,123 @@
-
-Click the default :guilabel:`🟰 (equal sign)` icon in the second portion of the filter equation. This
-reveals a drop-down menu. From this drop-down menu, select :guilabel:`contains`.
+- :guilabel:`Target`: defines the model of the target audience for the campaign. The assigned model
+  determines the fields that are available throughout the campaign, including the fields available
+  in the :guilabel:`Filter` section, and in dynamic placeholders.
```
- would add guis here since we're doing form coverage
- take earlier definition we removed and put it here since it's more digestible and easier to understand for folks

```suggestion
- :guilabel:`Target`: specifies the type of records available for use in the campaign, such as
  :guilabel:`Lead/Opportunity`, :guilabel:`Event Registration`, :guilabel:`Contact`, The assigned
  records model determines the fields that are available throughout the campaign, including the
  fields available in the :guilabel:`Filter` section, and in dynamic placeholders.
```

StraubCreative commented on content/applications/marketing/marketing_automation/target_audience.rst:
```diff
@@ -1,123 +1,123 @@
+- :guilabel:`Save as Favorite Filter`: saves the current :guilabel:`Filter` for future use with the
+  current :guilabel:`Target` model, and can be managed from the :menuselection:`Marketing Automation
+  app --> Configuration --> Favorite Filters` menu.
+- :guilabel:`Unicity based on`: specifies the :guilabel:`Target` model field at which duplicates
```
wording
```suggestion
- :guilabel:`Unicity based on`: specifies the :guilabel:`Target` model field where duplicates
```

StraubCreative commented on content/applications/marketing/marketing_automation/target_audience.rst:
```diff
@@ -1,123 +1,123 @@
+- :guilabel:`Unicity based on`: specifies the :guilabel:`Target` model field at which duplicates
+  should be avoided. Traditionally, the :guilabel:`Email` field is used, but any available field can
+  be used.
+- :guilabel:`Filter`: contains an interactive form to refine the target audience on a specific
+  subset of the :guilabel:`Target` model, detailed in the
+  :ref:`marketing_automation/defining-filters` section.
```
"Specific subset of the  **Target** model" is likely going to confuse non-technical people so here's a suggested rewrite:

```suggestion
- :guilabel:`Filter`: contains an interactive form with configurable logic to further refine the
  targeting parameters under the chosen :guilabel:`Target` model. See more details in the
  :ref:`marketing_automation/defining-filters` section.
```

StraubCreative commented on content/applications/marketing/marketing_automation/target_audience.rst:
```diff
@@ -1,123 +1,123 @@
-Then, add another node to this filter by clicking the :guilabel:`➕ (plus sign)` icon to the right of
-the equation.
+.. note::
+   Each activity in a campaign's workflow can target a subset of the target audience, see the
+   :doc:`workflow_activities` documentation for more information.
```
run-on sentence
```suggestion
   Each activity in a campaign's workflow can target a subset of the target audience; see the
   :doc:`workflow_activities` documentation for more information.
```

StraubCreative commented on content/applications/marketing/marketing_automation/target_audience.rst:
```diff
@@ -1,123 +1,123 @@
-name of the ticket type that should be used for the filter. In this case, :guilabel:`Standard` is
-the name of the event ticket type for this sample filter.
+To refine the :guilabel:`Filter` rules of a campaign, click the :guilabel:`➕ Add condition` button
+to reveal a new filter rule row.
```
break up the [noun stack](https://www.enago.com/academy/noun-stacks-why-you-should-avoid-them-in-scientific-writing/)
```suggestion
to reveal a new row with configurable rule parameters.
```

StraubCreative reviewed (approved):
Hi @samueljlieber 
Great doc! I really like the example you used at the end to tie it all together. 

Approving with comments— I have some suggestions for you to consider below, however we can merge whenever you're ready. 

Thanks for your contribution! 🚀

samueljlieber commented:
Thank you @StraubCreative for your insightful review! I implemented your suggestions in 6b187bf. I also removed a good portion of the configuring filters documentation since it is being improved in https://github.com/odoo/documentation/pull/8472, I added a reference to it instead.

samueljlieber commented:
@robodoo r+ please :)
